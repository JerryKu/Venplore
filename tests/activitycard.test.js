'use strict';

import React from 'react';
import FeaturedActivityCard from '../client/src/components/landingPage/FeaturedActivityCard';
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const tree = renderer
    .create(<FeaturedActivityCard
    image='http://ichef.bbci.co.uk/wwfeatures/wm/live/1280_640/images/live/p0/51/v6/p051v6vn.jpg'
    name='Fishing'
    description='Sun tanning or simply tanning is the process whereby skin color is darkened or tanned. It is most often a result of exposure to ultraviolet (UV) radiation from sunlight or from artificial sources'/>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('displays a placeholder image if there is none', () =>{
  const tree = renderer
  .create(<FeaturedActivityCard
    image = ''
    name='Fishing'
    description='Sun tanning or simply tanning is the process whereby skin color is darkened or tanned. It is most often a result of exposure to ultraviolet (UV) radiation from sunlight or from artificial sources'
    />).toJSON();
    expect(tree).toMatchSnapshot();
})
