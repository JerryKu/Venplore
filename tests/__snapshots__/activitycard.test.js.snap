// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`displays a placeholder image if there is none 1`] = `
<div
  className="activity-card absolute"
>
  <img
    className="activity-card-img"
    src="http://www.independentmediators.co.uk/wp-content/uploads/2016/02/placeholder-image-768x576.jpg"
  />
  <div
    className="activity-card-hover relative"
  >
    <div
      className="activity-card-description"
    >
      <p
        className="activity-card-text"
      >
        Sun tanning or simply tanning is the process whereby skin color is darkened or tanned. It is most often a result of exposure to ultraviolet (UV) radia...
      </p>
    </div>
    <div
      className="activity-card-title"
    >
      <h2>
        Fishing
      </h2>
    </div>
  </div>
</div>
`;

exports[`renders correctly 1`] = `
<div
  className="activity-card absolute"
>
  <img
    className="activity-card-img"
    src="http://ichef.bbci.co.uk/wwfeatures/wm/live/1280_640/images/live/p0/51/v6/p051v6vn.jpg"
  />
  <div
    className="activity-card-hover relative"
  >
    <div
      className="activity-card-description"
    >
      <p
        className="activity-card-text"
      >
        Sun tanning or simply tanning is the process whereby skin color is darkened or tanned. It is most often a result of exposure to ultraviolet (UV) radia...
      </p>
    </div>
    <div
      className="activity-card-title"
    >
      <h2>
        Fishing
      </h2>
    </div>
  </div>
</div>
`;
